### ISSUE TITLE
epochs

### ISSUE THREAD

**Issue by wyx19990420** [xiaojiu143/AF-YOLO#1]
Does the paper specify that the other YOLO methods were trained for 500 epochs?I trained YOLOv8n for 500 epochs using your parameters, but I didn’t achieve such high accuracy. Here are my test results，map0.5 (%) = 0.279, map0.5:0.95 (%) = 0.157; val: map0.5 (%) = 35.1, map0.5:0.95 (%) = 0.205. Did you enable the P2 layer in the baseline model for YOLOv8n? Could you help me resolve this issue? Thank you very much.

<img width="460" height="156" alt="Image" src="https://github.com/user-attachments/assets/455a0d88-f184-4057-aae2-aaf3da339c42" />

**Comment by zzwyx99** [comment]
I seem to be having some issues with training. Could you help me figure this out? These are the results from training for 500 epochs using the hyperparameters from the paper. Neither the baseline nor AF-YOLO is achieving the claimed accuracy. I tested adding a P2 detection head on YOLO8-11-26 and found that it improved accuracy, but removing P5 caused the accuracy to drop. Here are my experimental data; if you need them, I can also provide the `results.csv` file generated during training.
YOLOv8n-P2： test: map0.5 (%) = 0.304, map0.5:0.95 (%) = 0.174; val: map0.5 (%) = 0.381, map0.5:0.95 (%) = 0.229;
YOLOv8n-P2-withoutP5： test: map0.5 (%) = 0.293, map0.5:0.95 (%) = 0.168; val: map0.5 (%) = 0.374, map0.5:0.95 (%) = 0.224;
AF-YOLO: test: map0.5 (%) = 0.288, map0.5:0.95 (%) = 0.163; val: map0.5 (%) = 0.365, map0.5:0.95 (%) = 0.219;

**Comment by xiaojiu143** [comment]
> 论文里有没有说明其他YOLO方法是针对500个纪元训练的？我用你的参数训练了YOLOv8n，针对500个纪元，但精度没达到那么高。这是我的测试结果，map0.5 （%） = 0.279，map0.5：0.95 （%） = 0.157;val： map0.5 （%） = 35.1，map0.5：0.95 （%） = 0.205。你在YOLOv8n基线模型中启用了P2层吗？你能帮我解决这个问题吗？非常感谢。
> 
> <img alt="图片" width="460" height="156" src="https://private-user-images.githubusercontent.com/91590387/570133155-455a0d88-f184-4057-aae2-aaf3da339c42.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzQ5NDcwNjEsIm5iZiI6MTc3NDk0Njc2MSwicGF0aCI6Ii85MTU5MDM4Ny81NzAxMzMxNTUtNDU1YTBkODgtZjE4NC00MDU3LWFhZTItYWFmM2RhMzM5YzQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAzMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMzMxVDA4NDYwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNjMDhiNWNlMTlkMGYwMGYwYmZjNWY4ZWUxMWU5OTA1ZmFkYWJmMGQwMzhhZDdjMzlmZGRkNjVkODk2YmE2NWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.GJTttgKRoQusX33j3GI0qsoflpbSmt7_whcCyZxWo5o">

1. Due to differences in training hardware, it is recommended to increase the epochs during training (a setting of 2000 or higher is advised) and enable the early stopping mechanism (following Ultralytics’ recommendations, a patience value of 50 is suggested; training will terminate automatically upon convergence). This helps verify the convergence epochs applicable to your hardware configuration, ensuring the training results reach a fitted state to achieve optimal accuracy metrics.

2. The baseline models YOLOv8n-P2 and YOLOv8n-P2-withoutP5 were not evaluated in the paper. If you need to investigate the impact of different detection layers on the performance of PAFFN, please refer to Table VI in the original paper for relevant experiments.

3. In view of the issues you have encountered, screenshots of the model prediction results are provided for your reference to facilitate your subsequent research work.

<img width="1372" height="292" alt="Image" src="https://github.com/user-attachments/assets/29745083-47d8-469d-9c67-2083ec1819b8" />

<img width="1392" height="292" alt="Image" src="https://github.com/user-attachments/assets/c57bcfa4-d797-417b-91b6-206fb7104320" />

**Comment by zzwyx99** [comment]
> > 论文里有没有说明其他YOLO方法是针对500个纪元训练的？我用你的参数训练了YOLOv8n，针对500个纪元，但精度没达到那么高。这是我的测试结果，map0.5 （%） = 0.279，map0.5：0.95 （%） = 0.157;val： map0.5 （%） = 35.1，map0.5：0.95 （%） = 0.205。你在YOLOv8n基线模型中启用了P2层吗？你能帮我解决这个问题吗？非常感谢。
> > <img alt="图片" width="460" height="156" src="https://private-user-images.githubusercontent.com/91590387/570133155-455a0d88-f184-4057-aae2-aaf3da339c42.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzQ5NDcwNjEsIm5iZiI6MTc3NDk0Njc2MSwicGF0aCI6Ii85MTU5MDM4Ny81NzAxMzMxNTUtNDU1YTBkODgtZjE4NC00MDU3LWFhZTItYWFmM2RhMzM5YzQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNjAzMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjYwMzMxVDA4NDYwMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNjMDhiNWNlMTlkMGYwMGYwYmZjNWY4ZWUxMWU5OTA1ZmFkYWJmMGQwMzhhZDdjMzlmZGRkNjVkODk2YmE2NWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.GJTttgKRoQusX33j3GI0qsoflpbSmt7_whcCyZxWo5o">
> 
>     1. Due to differences in training hardware, it is recommended to increase the epochs during training (a setting of 2000 or higher is advised) and enable the early stopping mechanism (following Ultralytics’ recommendations, a patience value of 50 is suggested; training will terminate automatically upon convergence). This helps verify the convergence epochs applicable to your hardware configuration, ensuring the training results reach a fitted state to achieve optimal accuracy metrics.
> 
>     2. The baseline models YOLOv8n-P2 and YOLOv8n-P2-withoutP5 were not evaluated in the paper. If you need to investigate the impact of different detection layers on the performance of PAFFN, please refer to Table VI in the original paper for relevant experiments.
> 
>     3. In view of the issues you have encountered, screenshots of the model prediction results are provided for your reference to facilitate your subsequent research work.
> 
> 
> <img alt="Image" width="1372" height="292" src="https://private-user-images.githubusercontent.com/107339025/571781219-29745083-47d8-469d-9c67-2083ec1819b8.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzQ5NDg3MzIsIm5iZiI6MTc3NDk0ODQzMiwicGF0aCI6Ii8xMDczMzkwMjUvNTcxNzgxMjE5LTI5NzQ1MDgzLTQ3ZDgtNDY5ZC05YzY3LTIwODNlYzE4MTliOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMzMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDMzMVQwOTEzNTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lZWNmNTJhODUwOTFkMTlkNWMwMDIyOGNjNDc4NmYxNWQ1NWNkODQ3OTIzZjBkZjM0ZTQxZTk5YjIwYTBlOTRhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.VrVGmhVUQqpQmBzdvmOr28BvNyS84Oyf7AXsXeQ_6Xg"> <img alt="Image" width="1392" height="292" src="https://private-user-images.githubusercontent.com/107339025/571781327-c57bcfa4-d797-417b-91b6-206fb7104320.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzQ5NDg3MzIsIm5iZiI6MTc3NDk0ODQzMiwicGF0aCI6Ii8xMDczMzkwMjUvNTcxNzgxMzI3LWM1N2JjZmE0LWQ3OTctNDE3Yi05MWI2LTIwNmZiNzEwNDMyMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwMzMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDMzMVQwOTEzNTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNDliMmIxNTQ4OTYxNWI3YWY5MjJiNzBkMmFhMDU5NWIyZTYwNTcyNmYwZGYzMzAyZWE3ZDU5MzVhNGE2MzA1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.wjBFjcd40K5rLMFOoF7dIko-OK3ZyGEYD3eJeVZMk-o">

感谢回复，原来这个是在val上的结果，我之前认为是在test数据集上的结果了，那精度不会有显著差异。
